metadata-ingestion/src/datahub/ingestion/source/kafka.py

```diff
@@ -265,10 +304,71 @@ def _extract_record(self, topic: str) -> Iterable[MetadataWorkUnit]:
                 self.report.report_workunit(wu)
                 yield wu
 
+    def build_custom_properties(self, topic_detail, extra_topic_config):
+
+        replication_factor = None
+        for _, p_meta in topic_detail.partitions.items():
```

Comment:
If topic_detail.partitions or topic_detail.partitions.items() ever going to be None? Do we need to handle this case?

Reply:
Should not be. but updated code to handle the cases to be safe.